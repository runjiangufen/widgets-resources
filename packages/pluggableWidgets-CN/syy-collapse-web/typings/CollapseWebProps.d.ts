/**
 * This file was generated from CollapseWeb.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ComponentType, CSSProperties, ReactNode } from "react";
import { ActionValue } from "mendix";

export type CollapsibleEnum = "header" | "disabled";

export type ExpandIconPositionEnum = "start" | "end";

export interface CollapseWebContainerProps {
    name: string;
    class: string;
    style?: CSSProperties;
    tabIndex?: number;
    children?: ReactNode;
    accordion: boolean;
    bordered: boolean;
    collapsible: CollapsibleEnum;
    expandIconPosition: ExpandIconPositionEnum;
    ghost: boolean;
    onChangeAction?: ActionValue;
}

export interface CollapseWebPreviewProps {
    className: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    children: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    accordion: boolean;
    bordered: boolean;
    collapsible: CollapsibleEnum;
    expandIconPosition: ExpandIconPositionEnum;
    ghost: boolean;
    onChangeAction: {} | null;
}
