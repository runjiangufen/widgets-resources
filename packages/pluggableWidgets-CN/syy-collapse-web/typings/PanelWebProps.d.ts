/**
 * This file was generated from PanelWeb.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ComponentType, CSSProperties, ReactNode } from "react";

export type CollapsibleEnum = "header" | "disabled" | "undefined";

export interface PanelWebContainerProps {
    name: string;
    class: string;
    style?: CSSProperties;
    tabIndex?: number;
    children?: ReactNode;
    key: string;
    collapsible: CollapsibleEnum;
    extra?: ReactNode;
    forceRender: boolean;
    header: ReactNode;
    showArrow: boolean;
}

export interface PanelWebPreviewProps {
    className: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    children: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    key: string;
    collapsible: CollapsibleEnum;
    extra: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    forceRender: boolean;
    header: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    showArrow: boolean;
}
