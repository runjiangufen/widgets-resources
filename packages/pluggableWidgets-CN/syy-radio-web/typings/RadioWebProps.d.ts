/**
 * This file was generated from RadioWeb.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ActionValue, DynamicValue, EditableValue, ListValue, ListAttributeValue } from "mendix";
import { Big } from "big.js";

export type DatasTypeEnum = "dynamic" | "static";

export interface StaticDataType {
    manualValue: string;
    manuaTitle: string;
    manualDisabled: boolean;
}

export type SizeEnum = "large" | "middle" | "small";

export type ButtonStyleEnum = "outline" | "solid";

export type OptionTypeEnum = "default" | "button";

export type DirectionEnum = "horizontal" | "vertical";

export interface StaticDataPreviewType {
    manualValue: string;
    manuaTitle: string;
    manualDisabled: boolean;
}

export interface RadioWebContainerProps {
    name: string;
    tabIndex?: number;
    id: string;
    datasType: DatasTypeEnum;
    staticData: StaticDataType[];
    data?: ListValue;
    attribute?: ListAttributeValue<string | Date | Big>;
    title?: ListAttributeValue<string>;
    attributedata?: EditableValue<string>;
    selectedValue?: EditableValue<string | Date | Big>;
    isText: boolean;
    size: SizeEnum;
    buttonStyle: ButtonStyleEnum;
    disabled: DynamicValue<boolean>;
    radioName: string;
    optionType: OptionTypeEnum;
    direction: DirectionEnum;
    onChangeAction?: ActionValue;
    customWidgetOnChangeValidate: boolean;
}

export interface RadioWebPreviewProps {
    readOnly: boolean;
    datasType: DatasTypeEnum;
    staticData: StaticDataPreviewType[];
    data: {} | { type: string } | null;
    attribute: string;
    title: string;
    attributedata: string;
    selectedValue: string;
    isText: boolean;
    size: SizeEnum;
    buttonStyle: ButtonStyleEnum;
    disabled: string;
    radioName: string;
    optionType: OptionTypeEnum;
    direction: DirectionEnum;
    onChangeAction: {} | null;
    customWidgetOnChangeValidate: boolean;
}
