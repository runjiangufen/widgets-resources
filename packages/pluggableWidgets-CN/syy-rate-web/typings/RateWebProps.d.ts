/**
 * This file was generated from RateWeb.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ActionValue, DynamicValue, EditableValue } from "mendix";
import { Big } from "big.js";

export interface RateWebContainerProps {
    name: string;
    tabIndex?: number;
    id: string;
    value: EditableValue<Big>;
    allowClear: boolean;
    allowHalf: boolean;
    count: number;
    disabled: DynamicValue<boolean>;
    onChange?: ActionValue;
    customWidgetOnChangeValidate: boolean;
}

export interface RateWebPreviewProps {
    readOnly: boolean;
    value: string;
    allowClear: boolean;
    allowHalf: boolean;
    count: number | null;
    disabled: string;
    onChange: {} | null;
    customWidgetOnChangeValidate: boolean;
}
