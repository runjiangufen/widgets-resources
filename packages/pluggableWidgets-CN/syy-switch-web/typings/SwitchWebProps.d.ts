/**
 * This file was generated from SwitchWeb.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ActionValue, DynamicValue, EditableValue } from "mendix";

export type SizeEnum = "default" | "small";

export type ConfirmTypeEnum = "pop" | "modal";

export type ConfirmokTypeEnum = "primary" | "ghost" | "dashed" | "link" | "text" | "default";

export interface SwitchWebContainerProps {
    name: string;
    tabIndex?: number;
    id: string;
    checked: EditableValue<boolean>;
    onName: string;
    offName: string;
    loadingFlag?: EditableValue<boolean>;
    disabled?: EditableValue<boolean>;
    size: SizeEnum;
    autoFocus: boolean;
    onChange?: ActionValue;
    openconfirm: boolean;
    confirmType: ConfirmTypeEnum;
    ontitle?: DynamicValue<string>;
    offtitle?: DynamicValue<string>;
    confirmokText: string;
    confirmcancelText: string;
    confirmokType: ConfirmokTypeEnum;
    onConfirm?: ActionValue;
    onCancel?: ActionValue;
}

export interface SwitchWebPreviewProps {
    readOnly: boolean;
    checked: string;
    onName: string;
    offName: string;
    loadingFlag: string;
    disabled: string;
    size: SizeEnum;
    autoFocus: boolean;
    onChange: {} | null;
    openconfirm: boolean;
    confirmType: ConfirmTypeEnum;
    ontitle: string;
    offtitle: string;
    confirmokText: string;
    confirmcancelText: string;
    confirmokType: ConfirmokTypeEnum;
    onConfirm: {} | null;
    onCancel: {} | null;
}
