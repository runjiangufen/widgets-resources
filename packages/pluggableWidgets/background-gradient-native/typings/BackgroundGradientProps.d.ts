/**
 * This file was generated from BackgroundGradient.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ComponentType, CSSProperties, ReactNode } from "react";
import { ActionValue } from "mendix";
import { Big } from "big.js";

export interface ColorListType {
    color: string;
    offset: Big;
}

export interface ColorListPreviewType {
    color: string;
    offset: number | null;
}

export interface BackgroundGradientProps<Style> {
    name: string;
    style: Style[];
    colorList: ColorListType[];
    content: ReactNode;
    onClick?: ActionValue;
}

export interface BackgroundGradientPreviewProps {
    className: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    colorList: ColorListPreviewType[];
    content: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    onClick: {} | null;
}
