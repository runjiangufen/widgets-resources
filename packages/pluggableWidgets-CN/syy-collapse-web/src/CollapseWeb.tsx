import { ReactElement, createElement, useCallback } from "react";
import { CollapseWebContainerProps } from "../typings/CollapseWebProps";
import { PanelWebContainerProps } from "../typings/PanelWebProps";
import { executeAction } from "@mendix/piw-utils-internal";
import "./ui/index.css";
import { Collapse } from "antd";
import "antd/lib/collapse/style/css.js";
// import { Icon } from "./components/Icon";
const { Panel } = Collapse;

export default function CollapseWeb(props: CollapseWebContainerProps): ReactElement {
    const {
        accordion,
        bordered,
        collapsible,
        // expandIcon,
        expandIconPosition,
        ghost
    } = props;
    const handlerChangeAction = useCallback(() => executeAction(props.onChangeAction), [props.onChangeAction]);
    return (
        <Collapse
            className={`syy-collapse ${props.class}`}
            style={props.style}
            accordion={accordion}
            bordered={bordered}
            collapsible={collapsible}
            //  图标报错
            // expandIcon={expandIcon && expandIcon.value ? <Icon value={expandIcon.value} empty /> : undefined}
            expandIconPosition={expandIconPosition}
            ghost={ghost}
            onChange={handlerChangeAction}
        >
            {Array.isArray(props.children)
                ? props.children.map(
                      ({ props: childrenProps, key }: { props: PanelWebContainerProps; key: string }) => {
                          return (
                              <Panel
                                  className={childrenProps?.class}
                                  style={childrenProps?.style}
                                  showArrow={childrenProps?.showArrow}
                                  forceRender={childrenProps?.forceRender}
                                  extra={childrenProps?.extra}
                                  collapsible={
                                      childrenProps.collapsible === "undefined" ? undefined : childrenProps.collapsible
                                  }
                                  header={childrenProps?.header}
                                  key={key}
                              >
                                  {childrenProps.children}
                              </Panel>
                          );
                      }
                  )
                : null}
        </Collapse>
    );
}
