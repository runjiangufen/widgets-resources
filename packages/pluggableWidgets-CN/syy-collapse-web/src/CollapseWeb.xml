<?xml version="1.0" encoding="utf-8" ?>
<widget id="runjian.collapseweb.CollapseWeb" pluginWidget="true" needsEntityContext="true" offlineCapable="true" supportedPlatform="Web" xmlns="http://www.mendix.com/widget/1.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../node_modules/mendix/custom_widget.xsd">
    <name>SYY Collapse Web</name>
    <description>A content area that can be collapsed/expanded.</description>
    <icon />
    <properties>
        <propertyGroup caption="General">
            <property key="children" type="widgets" required="false">
                <caption>Children</caption>
                <description>Children</description>
            </property>
            <property key="accordion" type="boolean" defaultValue="false">
                <caption>Accordion</caption>
                <description>Accordion mode</description>
            </property>
            <property key="bordered" type="boolean" defaultValue="true">
                <caption>Bordered</caption>
                <description>Folding panel with border style</description>
            </property>
            <!-- <property key="defaultActiveKey" type="boolean" required="false">
                <caption>DefaultActiveKey</caption>
                <description>Initializes the key of the selected panel</description>
                <attributeTypes>
                    <attributeType></attributeType>                
                </attributeTypes>
            </property> -->
            <property key="collapsible" type="enumeration" defaultValue="header" required="true">
                <caption>Collapsible</caption>
                <description>Whether all subpanels are collapsible or specify collapsible trigger areas</description>
                <enumerationValues>
                    <enumerationValue key="header">header</enumerationValue>
                    <enumerationValue key="disabled">disabled</enumerationValue>
                </enumerationValues>
            </property>
            <!-- <property key="expandIcon" type="icon" required="false">
                <caption>expandIcon</caption>
                <description>设置按钮的图标- 暂时不支持</description>
            </property> -->
            <property key="expandIconPosition" type="enumeration" defaultValue="start">
                <caption>ExpandIconPosition</caption>
                <description>Setting icon location</description>
                <enumerationValues>
                    <enumerationValue key="start">start</enumerationValue>
                    <enumerationValue key="end">end</enumerationValue>
                </enumerationValues>
            </property>
            <property key="ghost" type="boolean" defaultValue="false">
                <caption>Ghost</caption>
                <description>Make the folded panel transparent and bezel-free</description>
            </property>
        </propertyGroup>
        <propertyGroup caption="Events">
            <property key="onChangeAction" type="action">
                <caption>OnChange</caption>
                <description>Callbacks when clicking on the panel</description>
            </property>
        </propertyGroup>
    </properties>
</widget>
