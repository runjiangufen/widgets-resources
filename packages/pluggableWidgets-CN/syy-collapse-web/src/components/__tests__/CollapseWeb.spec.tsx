import { createElement, ReactNode } from "react";
import { render } from "enzyme";
import CollapseWeb from "../../CollapseWeb";
import { Collapse } from "antd";
import { CollapseWebContainerProps, ExpandIconPositionEnum, CollapsibleEnum } from "../../../typings/CollapseWebProps";
import { PanelWebContainerProps } from "../../../typings/PanelWebProps";
const { Panel } = Collapse;

const COLLAPSIBLE_ENUM: CollapsibleEnum[] = ["header", "disabled"];
const EXPAND_ICON_POSITON_ENUM: ExpandIconPositionEnum[] = ["start", "end"];

describe("CollapseWeb", () => {
    let defaultCollapseWebProps: CollapseWebContainerProps,
        defaultPanelWebProps1: PanelWebContainerProps,
        defaultPanelWebProps2: PanelWebContainerProps;

    beforeEach(() => {
        defaultCollapseWebProps = {
            name: "",
            class: "",
            accordion: false,
            bordered: false,
            collapsible: COLLAPSIBLE_ENUM[0],
            expandIconPosition: EXPAND_ICON_POSITON_ENUM[0],
            ghost: false
        };
        defaultPanelWebProps1 = {
            name: "",
            class: "",
            key: "panel1",
            collapsible: COLLAPSIBLE_ENUM[0],
            forceRender: false,
            showArrow: false,
            header: "title1",
            children: "content1"
        };
        defaultPanelWebProps2 = {
            name: "",
            class: "",
            key: "panel2",
            collapsible: COLLAPSIBLE_ENUM[0],
            forceRender: false,
            showArrow: false,
            header: "title2",
            children: "content2"
        };
    });

    it("renders default collapseWeb", () => {
        const collapseWeb = render(<CollapseWeb {...defaultCollapseWebProps} />);
        expect(collapseWeb).toMatchSnapshot();
    });

    it("renders default collapseWeb and panel", () => {
        const collapseWeb = render(
            <CollapseWeb {...defaultCollapseWebProps}>
                <Panel {...defaultPanelWebProps1} />
                <Panel {...defaultPanelWebProps2} />
            </CollapseWeb>
        );
        expect(collapseWeb.find(".ant-collapse-item").length).toEqual(2);
        expect(collapseWeb).toMatchSnapshot();
    });

    // it("triggers onClick and OnChange function with a click event", () => {
    //     const onChangeAction = actionValue();
    //     const collapseWeb = mount(<CollapseWeb {...defaultCollapseWebProps} onChangeAction={onChangeAction}>
    //         <Panel {...defaultPanelWebProps1} />
    //         <Panel {...defaultPanelWebProps2}/>
    //     </CollapseWeb>);
    //     console.log(collapseWeb.find(".ant-collapse-header"))
    //     collapseWeb.find(".ant-collapse-header").at(0).simulate("click")
    //     expect(onChangeAction.execute).toHaveBeenCalledTimes(1);
    // });

    it("Set the class of collapseWeb", () => {
        const collapseWeb: ReactNode = render(<CollapseWeb {...defaultCollapseWebProps} class="testClass" />);
        expect(collapseWeb).toMatchSnapshot();
    });

    it("Set the bordered of collapseWeb", () => {
        const collapseWeb: ReactNode = render(<CollapseWeb {...defaultCollapseWebProps} bordered={true} />);
        expect(collapseWeb).toMatchSnapshot();
    });

    COLLAPSIBLE_ENUM.map(value => {
        it("Set the collapsible of collapseWeb to " + value, () => {
            const collapseWeb: ReactNode = render(<CollapseWeb {...defaultCollapseWebProps} collapsible={value} />);
            expect(collapseWeb).toMatchSnapshot();
        });
    });

    EXPAND_ICON_POSITON_ENUM.map(value => {
        it("Set the collapsible of collapseWeb to " + value, () => {
            const collapseWeb: ReactNode = render(
                <CollapseWeb {...defaultCollapseWebProps} expandIconPosition={value} />
            );
            expect(collapseWeb).toMatchSnapshot();
        });
    });

    it("Set the ghost of collapseWeb", () => {
        const collapseWeb: ReactNode = render(<CollapseWeb {...defaultCollapseWebProps} ghost={true} />);
        expect(collapseWeb).toMatchSnapshot();
    });
});
