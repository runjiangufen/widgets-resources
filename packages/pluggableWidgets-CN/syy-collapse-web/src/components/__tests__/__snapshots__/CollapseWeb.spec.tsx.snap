// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapseWeb Set the bordered of collapseWeb 1`] = `
<div
  class="ant-collapse ant-collapse-icon-position-start syy-collapse "
/>
`;

exports[`CollapseWeb Set the class of collapseWeb 1`] = `
<div
  class="ant-collapse ant-collapse-icon-position-start ant-collapse-borderless syy-collapse testClass"
/>
`;

exports[`CollapseWeb Set the collapsible of collapseWeb to disabled 1`] = `
<div
  class="ant-collapse ant-collapse-icon-position-start ant-collapse-borderless syy-collapse "
/>
`;

exports[`CollapseWeb Set the collapsible of collapseWeb to end 1`] = `
<div
  class="ant-collapse ant-collapse-icon-position-end ant-collapse-borderless syy-collapse "
/>
`;

exports[`CollapseWeb Set the collapsible of collapseWeb to header 1`] = `
<div
  class="ant-collapse ant-collapse-icon-position-start ant-collapse-borderless syy-collapse "
/>
`;

exports[`CollapseWeb Set the collapsible of collapseWeb to start 1`] = `
<div
  class="ant-collapse ant-collapse-icon-position-start ant-collapse-borderless syy-collapse "
/>
`;

exports[`CollapseWeb Set the ghost of collapseWeb 1`] = `
<div
  class="ant-collapse ant-collapse-icon-position-start ant-collapse-borderless ant-collapse-ghost syy-collapse "
/>
`;

exports[`CollapseWeb renders default collapseWeb 1`] = `
<div
  class="ant-collapse ant-collapse-icon-position-start ant-collapse-borderless syy-collapse "
/>
`;

exports[`CollapseWeb renders default collapseWeb and panel 1`] = `
<div
  class="ant-collapse ant-collapse-icon-position-start ant-collapse-borderless syy-collapse "
>
  <div
    class="ant-collapse-item ant-collapse-no-arrow"
  >
    <div
      aria-disabled="false"
      aria-expanded="false"
      class="ant-collapse-header ant-collapse-header-collapsible-only"
    >
      <span
        class="ant-collapse-header-text"
      >
        title1
      </span>
    </div>
  </div>
  <div
    class="ant-collapse-item ant-collapse-no-arrow"
  >
    <div
      aria-disabled="false"
      aria-expanded="false"
      class="ant-collapse-header ant-collapse-header-collapsible-only"
    >
      <span
        class="ant-collapse-header-text"
      >
        title2
      </span>
    </div>
  </div>
</div>
`;
